hphp/runtime/ext/xdebug/ext_xdebug.cpp

```diff
@@ -482,8 +493,19 @@ static Array HHVM_FUNCTION(xdebug_get_code_coverage) {
         tmp.set(line.first(), Variant(CodeCoverage::kLineExecuted));
       }
     }
-    ret.set(report.first(), tmp);
+    coverage.set(report.first(), tmp);
+  }
+
+  auto arg = Array::Create();
+  arg.set(0, coverage);
+  auto const merged = HHVM_FN(array_replace_recursive)(executables, deads, arg);
+  Array ret = Array::Create();
+  for (ArrayIter iter(merged); iter; ++iter) {
+    Array tmp = iter.second().toArray();
+    tmp->ksort(SORT_REGULAR, true);
+    ret.set(iter.first(), tmp);
```

Comment:
This should all be able to be left the same.  Some filtering may need to happen for the DEAD or UNUSED values if the recording was being performed with it, then stopped without throwing away data, then restarted without DEAD and UNUSED data.  I don't quite know xdebug's exact behavior there, but it might not show the older DEAD or UNUSED values.